### ISSUE TITLE
Website: no comparison with Apache Commons Email

### ISSUE THREAD

**Issue by Chealer** [bbottema/simple-java-mail#422]
My team uses an outdated version of Apache Commons Email. While the obvious solution may be to upgrade, the status (and future) of Commons Email [seems questionable](https://issues.apache.org/jira/projects/EMAIL/issues/EMAIL-197), so I looked for alternatives and discovered Simple Java Mail.

I unfortunately could not find any comparison between the 2. The closest I could find is [Mailtrap's guide](https://mailtrap.io/blog/sending-email-using-java/). Given that Commons Email claims it aims to simplify the Java Mail API and that Simple Java Mail's very name defines it quite the same, it is far from clear which one is preferable.

While a full comparison of Java mail libraries which would also include Spring's offer, which may be the best reference for us, is probably out of scope, Simple Java Mail's website should offer some comparison with at least Commons Email, which is more established due to its backing by the ASF and its seniority. At least a list of advantages (perhaps greater flexibility, documentation, stability) and some comparison of how to achieve the same task with either Commons Email or Simple Java Mail would go a long way. Simple Java Mail claims to be...

> the easiest to use (Java) mailing library in the world for sending emails using SMTP.

...and should therefore be simpler than Commons Email.

**Comment by bbottema** [comment]
I suppose it is reasonable to offer some kind of comparison to other frameworks, but I'm not sure what that would look like. This library is extensively documented both user documentation and api documentation, I'm not sure what more you need.

Perhaps you are looking for a feature matrix?

**Comment by Chealer** [comment]
Thanks Benny,
This was a general report not asking for any specific solution. Having a feature matrix is particularly useful when comparing many options. As this is about comparing 2 or 3 options, a matrix may not be that relevant.

This could start as simple as an unordered list of advantages Simple Java Mail has over Commons Email (and perhaps a list of disadvantages).

The comparison should not be restricted to features, but can also include qualities and other properties like licenses. If both components have the same feature set but one is more complicated or risky to use than the other, those are also differences which are important to highlight. Even if some comparisons are subjective and could be biased, having something would be a start.

I will consider any reasonable comparison as a solution to this issue.

**Comment by bbottema** [comment]
So I finally took the time and sat down to properly compare the libraries and produce a comparison matrix. It was quite a lot of work, making sure my claims about other libraries are correct or at least as much in good faith as possible.

However, I discovered two things: 

1. There are virtually no other large or active libraries out there anymore.
2. Apache Commons Email seemingly has been abandoned and left pretty half baked compared to Simple Java Mail. Its last release was in 2017 and has had no meaningful code changes since.

You can have a look at my first try here: https://www.simplejavamail.org/feature-matrix.html#navigation

Let me know if this is useful to you.

**Comment by Chealer** [comment]
Thank you very much Benny! I have no doubt that researching and writing this must have taken a lot of time, but this is definitely going to help us! :-)

The result is already much better than what I imagined, but in case it helps, here are my remarks:
* "All RFC stuff hidden behind clean (fluent) API?": I understand what this means, but expect many users will have no idea about RFC-s. It may be clearer to title the row "API simplicity" and to have values like "Clean functional (and fluent) API" vs "Exposes RFC stuff"
* Similarly, I would rename "Excellent (searchable) documentation
with lots of examples?" to "Quality documentation" and use values to specify ("Excellent (searchable) with lots of examples")
* Saying that Commons Email does not "Integrates with Spring?" seems misleading considering that we are using both together. I understand what the line means, but it could be phrased more accurately.
* It is unclear to me what "Supports core complexity?" means. Does "core complexity" mean something particular with regard to email?
* It appears incorrect to state that Commons Email does not do _Email address validation_. We are using version 1.2 and it does validate the sender address. However, I should specify that the validation relies on JavaMail, which means the validation depends on the JavaMail implementation (we found that Apache Geronimo does not validate nearly as well as the reference implementation).

Thank you again!👌
